### ISSUE TITLE
Support for ShouldSerialize pattern

### ISSUE THREAD

**Issue by Tornhoof** [neuecc/Utf8Json#21]
Most C# serializers support the ShouldSerializeXXX pattern to programmatically decide if a value should be serialized.
Example from JSON.NET (https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm):
```csharp
public class Employee
{
    public string Name { get; set; }
    public Employee Manager { get; set; }

    public bool ShouldSerializeManager()
    {
        // don't serialize the Manager property if an employee is their own manager
        return (Manager != this);
    }
}
```
It is for example used quite often to not serialize empty collections, e.g.
```csharp
using System.Collections.Generic;

namespace ConsoleApp9
{
    class Program
    {
        static void Main(string[] args)
        {
            var test = new Test();
            var text = Utf8Json.JsonSerializer.ToJsonString(test);
        }
    }

    public class Test
    {
        public List<string> Data { get; set; } = new List<string>();

        public bool ShouldSerializeData()
        {
            return Data?.Count > 0;
        }
    }
}
```

It would be great if support for this could be added.

**Comment by neuecc** [comment]
thank you for your suggestion.
I've implemented it.

**Comment by neuecc** [comment]
I've released `v1.2.3`, supports it.

**Comment by Tornhoof** [comment]
Thanks, it works. closing this
